.travis-ci.sh

```diff
@@ -76,24 +93,31 @@ case "$TARGET" in
       opam install utop ppx_driver odoc ocaml-migrate-parsetree js_of_ocaml-ppx --yes
       echo -en "travis_fold:end:opam.deps\r"
     fi
-    echo -en "travis_fold:start:jbuilder.bootstrap\r"
-    ocaml bootstrap.ml
-    echo -en "travis_fold:end:jbuilder.bootstrap\r"
-    ./boot.exe --subst
-    echo -en "travis_fold:start:jbuilder.boot\r"
-    ./boot.exe --dev
-    echo -en "travis_fold:end:jbuilder.boot\r"
-    if [ $WITH_OPAM -eq 1 ] ; then
-      _build/install/default/bin/jbuilder runtest && \
-      _build/install/default/bin/jbuilder build @test/blackbox-tests/runtest-js && \
-      ! _build/install/default/bin/jbuilder build @test/fail-with-background-jobs-running
-      RESULT=$?
-      if [ $UPDATE_OPAM -eq 0 ] ; then
-        rm -rf ~/.opam
-        mv ~/.opam-start ~/.opam
+    for version in $OCAML_VERSIONS ; do
+      BRANCH=${version%.*}
+      PATH=~/ocaml/$BRANCH/bin:$INITIAL_PATH
+      export PATH
+      git clean -dfx
+      echo -en "travis_fold:start:jbuilder.bootstrap-$version\r"
+      ocaml bootstrap.ml
+      echo -en "travis_fold:end:jbuilder.bootstrap-$version\r"
+      ./boot.exe --subst
+      echo -en "travis_fold:start:jbuilder.boot-$version\r"
+      ./boot.exe --dev
+      echo -en "travis_fold:end:jbuilder.boot-$version\r"
+      if [[ -n $OPAM_OCAML_VERSION ]] ; then
+        eval $(opam config env)
+        _build/install/default/bin/jbuilder runtest && \
+        _build/install/default/bin/jbuilder build @test/blackbox-tests/runtest-js && \
```

Comment:
This has nothing to do with this change, but why do we single out the JS tests like this?